optimum/habana/accelerate/accelerator.py

```diff
@@ -15,619 +15,78 @@
-                        "zero_optimization.stage3_param_persistence_threshold": 10 * hidden_size,
-                    }
-                )
-
-            if isinstance(optimizer, (DummyOptim)):
-                config_kwargs.update(
-                    {"optimizer.params.lr": optimizer.lr, "optimizer.params.weight_decay": optimizer.weight_decay}
-                )
-            if isinstance(scheduler, (DummyScheduler)) and scheduler.lr_scheduler_callable is None:
-                max_lr = (
-                    getattr(scheduler.optimizer, "lr", None)
-                    if getattr(scheduler.optimizer, "defaults", None) is None
-                    else scheduler.optimizer.defaults["lr"]
-                )
-                config_kwargs.update(
-                    {
-                        "scheduler.params.warmup_min_lr": 0,
-                        "scheduler.params.warmup_max_lr": max_lr,
-                        "scheduler.params.warmup_num_steps": scheduler.warmup_num_steps,
-                    }
-                )
-                if scheduler.total_num_steps is not None:
-                    config_kwargs["scheduler.params.total_num_steps"] = (
-                        math.ceil(scheduler.total_num_steps / self.num_processes)
-                        if not self.split_batches
-                        else scheduler.total_num_steps
-                    )
-            deepspeed_plugin.deepspeed_config_process(must_match=False, **config_kwargs)
-            self.deepspeed_config = deepspeed_plugin.deepspeed_config
-            kwargs = {"model": model, "config_params": self.deepspeed_config}
-            if optimizer is not None:
-                if isinstance(optimizer, (DummyOptim)):
-                    kwargs["model_parameters"] = optimizer.params
-                    if isinstance(scheduler, (DummyScheduler)) and scheduler.lr_scheduler_callable is not None:
-                        kwargs["lr_scheduler"] = scheduler.lr_scheduler_callable
-                else:
-                    if self.deepspeed_config["zero_optimization"].get("offload_optimizer", {}).get(
-                        "device", "none"
-                    ) != "none" and self.deepspeed_config.get("zero_force_ds_cpu_optimizer", True):
-                        from deepspeed.ops.adam import DeepSpeedCPUAdam
-
-                        defaults = {k: v for k, v in optimizer.defaults.items() if k in ["lr", "weight_decay"]}
-                        optimizer = DeepSpeedCPUAdam(optimizer.param_groups, **defaults)
-                    kwargs["optimizer"] = optimizer
-                    if scheduler is not None:
-                        if type(scheduler).__name__ in deepspeed.runtime.lr_schedules.VALID_LR_SCHEDULES:
-                            kwargs["lr_scheduler"] = scheduler
+        orig_num_models = len(self._models)
+        tuple = super()._prepare_deepspeed(*args)
```

Comment:
```suggestion
        prepared_deepspeed = super()._prepare_deepspeed(*args)
```

Please use prepared_deepspeed or result instead of tuple to avoid confusion with the built-in type

Reply:
I tried to keep the nomenclature used in accelerate